subprojects/language-java/src/main/java/org/gradle/api/tasks/compile/CompileOptions.java

```diff
@@ -73,6 +76,8 @@
 
     private FileCollection sourcepath;
 
+    private DiagnosticListener<? super JavaFileObject> diagnosticListener;
```

Comment:
I am a bit concerned about exposing a Java-based interface that might change with specific versions of the JDK. We could probably rather expose an interface we introduce in the Gradle API that syntheses to the underlying Java interface.